### ISSUE TITLE
Alias issue with join + aggregation

### ISSUE THREAD

**Issue by bgranvea** [ClickHouse/ClickHouse#17168]
**How to reproduce**

```
root@bb6e3ce5ed6f:/# clickhouse-client
ClickHouse client version 20.11.2.1 (official build).
Connecting to localhost:9000 as user default.
Connected to ClickHouse server version 20.11.2 revision 54442.

bb6e3ce5ed6f :) SELECT c1.value,c0.value FROM (select 1 AS insid,2 AS ts,3 AS value) c1 LEFT JOIN (SELECT 1 AS insid,2 AS ts,3 AS value) c0 ON c1.insid=c0.insid AND c1.ts=c0.ts GROUP BY c0.value,c1.value

=> OK

bb6e3ce5ed6f :) SELECT max(c1.value),c0.value FROM (select 1 AS insid,2 AS ts,3 AS value) c1 LEFT JOIN (SELECT 1 AS insid,2 AS ts,3 AS value) c0 ON c1.insid=c0.insid AND c1.ts=c0.ts GROUP BY c0.value

=> error "Column `value` is not under aggregate function and not in GROUP BY: While processing value, c0.value."
```

There is a workaround: don't use the same name "value" in c0 and c1:

```
bb6e3ce5ed6f :) SELECT max(c1.v1),c0.v0 FROM (select 1 AS insid,2 AS ts,3 AS v1) c1 LEFT JOIN (SELECT 1 AS insid,2 AS ts,3 AS v0) c0 ON c1.insid=c0.insid AND c1.ts=c0.ts GROUP BY c0.v0                              

=> OK
```

**Comment by den-crane** [comment]
Worked before:

```
19.13.7.57

SELECT max(c1.value),c0.value FROM (select 1 AS insid,2 AS ts,3 AS value) c1 
 LEFT JOIN (SELECT 1 AS insid,2 AS ts,3 AS value) c0 ON c1.insid=c0.insid AND c1.ts=c0.ts 
GROUP BY c0.value
```

**Comment by CurtizJ** [comment]
Should be fixed by #16951.
